### PR TITLE
py-hid: new port

### PR DESCRIPTION
**Pull request by bashu**
#### Description

Python hidapi bindings in ctypes (aka pyhidapi)

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.6.2 21G320 x86_64
Xcode 14.1 14B47b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: python/py-hid/Portfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+homepage            https://github.com/apmorton/pyhidapi
+
+checksums           rmd160  8afa962c1e67e7d0f06e1cf9d28f8737e8f1e19e \
+                    sha256  1e954e7f7ab9b7c9dfc78db59504692c17db3b71249492b976b1525b97dbb0e8 \
+                    size    3881
+
+patchfiles          hid-__init__.py.patch
+
+python.versions     34 35 36 37 38 39 310
```
Comment (reneeotten) [inline python/py-hid/Portfile:27]: no subports for unsupported Python versions please - remove 34,35,36

File: python/py-hid/Portfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-hid
+version             1.0.5
+
```
Comment (reneeotten) [inline python/py-hid/Portfile:8]: add "revision 0" line please

File: python/py-hid/Portfile [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+python.versions     34 35 36 37 38 39 310
+
+if {${name} ne ${subport}} {
+    depends_build-append \
+        port:py${python.version}-setuptools
+
+    depends_lib     port:hidapi
+
+    livecheck.type  none
```
Comment (reneeotten) [inline python/py-hid/Portfile:35]: You can remove this line as it is already set by the python PG

File: python/py-hid/Portfile [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+patchfiles          hid-__init__.py.patch
+
+python.versions     34 35 36 37 38 39 310
+
+if {${name} ne ${subport}} {
+    depends_build-append \
+        port:py${python.version}-setuptools
+
+    depends_lib     port:hidapi
```
Comment (reneeotten) [inline python/py-hid/Portfile:33]: this should become "depends_lib-append" as not to overwrite the dependency already added by the python PG

**Review by reneeotten — CHANGES_REQUESTED** [review]
thanks for the submission @bashu, a few comments

File: python/py-hid/Portfile (hunk 1 above)
Comment (bashu) [reply python/py-hid/Portfile]: fixed

File: python/py-hid/Portfile (hunk 2 above)
Comment (bashu) [reply python/py-hid/Portfile:9]: fixed

File: python/py-hid/Portfile (hunk 3 above)
Comment (bashu) [reply python/py-hid/Portfile]: fixed

File: python/py-hid/Portfile (hunk 4 above)
Comment (bashu) [reply python/py-hid/Portfile]: fixed

File: python/py-hid/Portfile [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+PortGroup           python 1.0
+
+name                py-hid
+version             1.0.5
+revision            0
+
+categories-append   comms
+
+license             MIT
+supported_archs     noarch
```
Comment (reneeotten) [inline python/py-hid/Portfile:13]: from the .whl file on PyPI it looks like this port will install the same files on each macOS version  "platforms {darwin any}" could/should be added. 

Please check this in tour other submissions as well (wheel file contains "any_none").

**Review by reneeotten — CHANGES_REQUESTED** [review]

**Comment by reneeotten** [comment]
ping @bashu - do you want to address the comments on your recent py-* ports so that we can get them merged?

**Comment by bashu** [comment]
to @reneeotten yes I will fix them, I was busy last week

**Review by reneeotten — APPROVED** [review]

**Pull request merged**